### PR TITLE
AttributeConsumingService support.

### PR DESCRIPTION
**Pull request by jimmyislive**

### CODE REVIEW

File: src/onelogin/saml2/metadata.py [hunk 1]
```diff
@@ -119,7 +182,7 @@ def builder(sp, authnsign=False, wsign=False, valid_until=None, cache_duration=N
             str_contacts = '\n'.join(contacts_info)
 
         metadata = """<?xml version="1.0"?>
-<md:EntityDescriptor xmlns:md="urn:oasis:names:tc:SAML:2.0:metadata"
+<md:EntityDescriptor xmlns:md="urn:oasis:names:tc:SAML:2.0:metadata" xmlns:saml="urn:oasis:names:tc:SAML:2.0:assertion"
```
Comment (pitbulk) [inline src/onelogin/saml2/metadata.py]: Can we add xmlns:saml namespace here only if attribute_consuming_service defined?

Comment (jimmyislive) [inline src/onelogin/saml2/metadata.py]: Sure, i'll add that in.

File: tests/src/OneLogin/saml2_tests/authn_request_test.py [hunk 2]
```diff
@@ -255,7 +255,7 @@ def testCreateEncSAMLRequest(self):
         inflated = decompress(decoded, -15)
 
         self.assertRegexpMatches(inflated, '^<samlp:AuthnRequest')
-        self.assertRegexpMatches(inflated, 'AssertionConsumerServiceURL="http://stuff.com/endpoints/endpoints/acs.php">')
+        self.assertRegexpMatches(inflated, 'AssertionConsumerServiceURL="http://stuff.com/endpoints/endpoints/acs.php"(\s)*>')
```
Comment (pitbulk) [inline tests/src/OneLogin/saml2_tests/authn_request_test.py]: why this change is required?

Can you add more unit test to cover that new attributeConsumingService feature and adapt old tests?

Comment (jimmyislive) [inline tests/src/OneLogin/saml2_tests/authn_request_test.py]: The RegExp unit tests were breaking caus of the presence of a training whitespace in Circle-CI. The \s will ignore all whitespaces. (e.g. when attributeConsumingService is not there)

yeah...i have yet to add unit tests...

File: setup.py [hunk 3]
```diff
@@ -9,7 +9,7 @@
 
 setup(
     name='python-saml',
-    version='2.1.6',
+    version='2.2.0',
```
Comment (pitbulk) [inline setup.py]: Can you keep the 2.1.6 version.

We only modify that value when a release is made, and we just made 1 release.
We will release a new version once we have some new features.

**Comment by jimmyislive** [comment]
@pitbulk please review

added new unti tests - circle-ci now passes

**Comment by jimmyislive** [comment]
ping @pitbulk 

any further comments on this?

**Comment by pitbulk** [comment]
Hi @jimmyislive

I had no time to review it. It will take some time, right now focus on other tasks sorry.

**Comment by pitbulk** [comment]
At python3-saml we following a XML template strategy, but here we build the XML by concat strings.
I know is "more dirty" but is how currently is implemented
